unstructured/embed/openai.py

```diff
@@ -0,0 +1,57 @@
+from typing import List, Optional
+
+import numpy as np
+
+from unstructured.documents.elements import (
+    Element,
+)
+from unstructured.embed.interfaces import BaseEmbeddingEncoder
+from unstructured.ingest.error import EmbeddingEncoderConnectionError
+from unstructured.utils import requires_dependencies
+
+
+class OpenAIEmbeddingEncoder(BaseEmbeddingEncoder):
+    def __init__(self, api_key: str, model_name: Optional[str] = "text-embedding-ada-002"):
+        self.api_key = api_key
+        self.model_name = model_name
+        self.initialize()
+
+    def initialize(self):
+        self.openai_client = self.get_openai_client()
+
+    def num_of_dimensions(self):
+        return np.shape(self.examplary_embedding)
+
+    def is_unit_vector(self):
+        return np.isclose(np.linalg.norm(self.examplary_embedding), 1.0)
+
+    def embed_query(self, query):
+        return self.openai_client.embed_documents(str(query))
+
+    def embed_documents(self, elements: List[Element]) -> List[Element]:
+        embeddings = self.openai_client.embed_documents([str(e) for e in elements])
```

Comment:
Just curious, does the ordering of the elements affect the embeddings? Might want to sort them somehow if so.

Reply:
Oh wait elements are already sorted, ignore.